src/rtk/features/singleRights/singleRightsApi.ts

```diff
@@ -64,9 +62,22 @@ export const singleRightsApi = createApi({
     getResourceOwners: builder.query<ResourceOwner[], void>({
       query: () => 'resources/resourceowners',
     }),
+    clearAccessCashe: builder.mutation<void, { party: string; user: BaseAttribute }>({
```

Comment:
I don't understand why this is a mutation, we don't use any data from that call? So a simple axios call would do. It's not critical for release but anyways

Reply:
It's a mutation because it's a PUT call (and it technically does change something on backend, namely the cache)
I put it in the SingleRightsApi as an rtkQuery because it's related to the rest of the SingleRights feature and part of the BFF's API's so this way it adheres to our already established pattern.
Not to mention that it's about as easy to add it to the rtkQuery as it would have been to use an axios call (with the added benefit of abstracting the actual network call away from this file) 😊

Reply:
There's not an established pattern for this, as this is the first time we're doing cache invalidation in the frontend or BFF. Yes it is a HTTP request but different types of HTTP requests should have their own pattern.

The difference between this and the other HTTP requests we have in frontend, is that this request doesn't receive any object in return and hence I don't think it needs to be a whole mutation because if this had been written as axios-call instead it would have made it clear that this request doesn't receive any data related from the request (except of HTTPResponse, but every HTTP Request has HTTPResponse). If all our future HTTP requests that doesn't receive object had been axios calls it would have made it clear in code base that when we do axios calls we don't expect any data. Also axios is more lightweight than Redux (although when receiving object, redux is better)

And that is a pattern.

Reply:
If this on the other hand was a Put request receving an object from the request then yeah I agree it would make sense this being a mutation

Reply:
I don't understand...
Why create an entirely different standard for something that is already covered by an existing one in our setup?

When I'm talking about standard and pattern here, I mean how we do HTTP calls (GET, POST, PUT, DELETE) and our current way of doing that is through rtkQuery (or through rtkSlices, but we've talked about moving a lot of that over to rtkQuery anyway). It's organized, it's reusible, and adding something to an existing Api-file is easy as sugar. (And this specific endpoint is part of the Single-Rights API anyway, so it fits right into that same rtk api setup)

I just don't see how any of this would be affected by whether or not the call returns a body?